### ISSUE TITLE
Please consider support on py310

### ISSUE THREAD

**Issue by BirchKwok** [LAMDA-NJU/Deep-Forest#124]

**Comment by leschultz** [comment]
Yes, please add this support. If it is of use to anybody else, I  noticed that I can use pip to install deep-forest only for python3.8 installed through miniconda using linux. You will need to downgrade numpy to 1.19.5 because anything newer breaks.

**Comment by simonprovost** [comment]
> No description provided.

We've opened a community fork of `Deep Forest` to support `Python 3.10+` 🥳 mainly due to the fact that current maintainers are either on other avenues of research or too slowly maintaining the project for years; [it has kindly been tried to support 3.10+](https://github.com/LAMDA-NJU/Deep-Forest/pull/155) _back in September 2025_ [but with CI and CD failing ](https://github.com/LAMDA-NJU/Deep-Forest/actions/runs/17707533516) we decided to go ahead on a side-gig while making sure to acknowledge the primary authors’ amazing work on Deep Forest!

> [!NOTE]
> _PS: We made sure to mention that if a pull request is welcomed and that it will be treated as high priority we are more than welcome to do that and remove the community fork, but we've waited too long._

**Repo**: https://github.com/simonprovost/deep_forest_py310
**PyPI**: https://pypi.org/project/deep-forest-py310/

Cheers!
